app/models/message.rb

```diff
@@ -39,7 +40,11 @@ class Message < ApplicationRecord
     end
 
     event :mark_as_sent do
-      transitions from: :sending, to: :sent
+      transitions from: %i[sending sent], to: :sent
+    end
+
+    event :mark_as_delivered do
+      transitions from: %i[sending sent], to: :delivered
```

Comment:
From looking at the logs in the chat, it didn't transition from `sending` to `sent` first. It went from `sending` to `delivered`. But if we look at the SMPP spec, it might be possible to get the enroute first and get back another final stat later.